services/metadata_service/api/utils.py

```diff
@@ -53,10 +54,20 @@ def handle_exceptions(func):
     """Catch exceptions and return appropriate HTTP error."""
 
     @wraps(func)
-    async def wrapper(*args, **kwargs):
+    async def wrapper(self, request):
         try:
-            return await func(*args, **kwargs)
+            return await func(self, request)
         except Exception as err:
-            return http_500(str(err))
+            # either use provided traceback from subprocess, or generate trace from current process
+            err_trace = getattr(err, 'traceback_str', None) or get_traceback_str()
+            print(err_trace)
+            # We log the request that caused this for debugging information
+            try:
+                body = await request.text()
+            except:
+                body = '<no body>'
+            print("Error caused when %s %s with query %s and body %s" %
```

Comment:
same deal here, could be using the logger instead to adhere to configured log levels.

Reply:
Yep, will change to a logger here and elsewhere. I will try to get the proper one (I couldn't find it initially).